scripts/get-nightly-build-pass-rate.ts

```diff
@@ -0,0 +1,108 @@
+import puppeteer from 'puppeteer';
+import dotenv from 'dotenv';
+dotenv.config();
+
+const loginAndAnalyzeBuilds = async (
+  latestBuildNumber: number,
+  numberOfDays: number,
+  username: string,
+  token: string
+) => {
+  const loginUrl = process.env.JENKINS_LOGIN_URL;
+  const baseApiUrl = process.env.JENKINS_BASE_API_URL;
+  const browser = await puppeteer.launch({ headless: true });
+  const page = await browser.newPage();
+
+  if (!loginUrl || !baseApiUrl) {
+    console.error('Missing environment variables. Please check your .env file.');
+    return;
+  }
+
+  try {
+    console.log('Logging in to Jenkins...');
+    await page.goto(loginUrl, { waitUntil: 'networkidle2' });
+    await page.type('#j_username', username);
+    await page.type('input[name="j_password"]', token);
+    await Promise.all([
+      page.click('button[type="submit"]'),
+      page.waitForNavigation({ waitUntil: 'networkidle2' }),
+    ]);
+    console.log('Login successful.');
```

Comment:
Just checking but pretty much everything before here will be removed once you get access to the Jenkins REST API?

Reply:
Yes, that's the plan. Once we sort out the issue with API credentials, we can probably dispense with Puppeteer, which feels like a heavy-handed tool for what this script could be. At that point, we’d likely switch to something simpler, like `node-fetch`, to handle the requests.